darwin/future/core/datasets/list_datasets.py

```diff
@@ -0,0 +1,38 @@
+from typing import Optional
+
+from pydantic import parse_obj_as
+
+from darwin.future.core.client import Client
+from darwin.future.core.types import TeamSlug
+from darwin.future.core.types.common import QueryString
+from darwin.future.data_objects.dataset import Dataset, DatasetList
+
+
+def list_datasets(api_client: Client, team_slug: Optional[TeamSlug] = None) -> DatasetList:
+    """
+    Returns a list of datasets for the given team
+
+    Parameters
+    ----------
+    api_client : Client
+        The client to use to make the request
+    team_slug : Optional[TeamSlug]
+        The slug of the team to retrieve datasets for
+
+    Returns
+    -------
+    DatasetList
+
+    Raises
+    ------
+    HTTPError
+        Any errors that occurred while making the request
+    ValidationError
+        Any errors that occurred while parsing the response
+
+    """
+    response = (
+        api_client.get("/datasets", QueryString({"team": str(team_slug)})) if team_slug else api_client.get("/datasets")
+    )
+
+    return parse_obj_as(DatasetList, response)
```

Comment:
is parse_obj_as and DatasetLists.parse_obj functionally the same or is there some difference? I've been using the latter. If they're the same we should pick one and keep it standard.

Reply:
`DatasetLists` doesn't have a `parse` method in this instance, because its def is just (roughly)

```python
DatasetList = List[Dataset]
```

So it's not inherently pydantic-y.  `parse_obj_as` is allowing us to co-erce a list of a pydantic type into validator iteratively, essentially - that said, this PR is pre-test, so I'm not convinced of the approach.

Reply:
ah right I see the issue. I haven't created types for mine, I just leave them as List[DarwinObject] in the code